maven-extension/src/main/java/io/opentelemetry/maven/AutoConfigureUtil2.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.maven;
+
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/** Utility class to use the {@link AutoConfiguredOpenTelemetrySdk}. */
+public class AutoConfigureUtil2 {
+
+  private AutoConfigureUtil2() {}
+
+  /**
+   * Returns the {@link Resource} that was autoconfigured.
+   *
+   * <p>Inspired by {@link
+   * io.opentelemetry.sdk.autoconfigure.internal.AutoConfigureUtil#getConfig(AutoConfiguredOpenTelemetrySdk)}
+   */
+  public static Resource getResource(
```

Comment:
[minor] can you elaborate a bit why do we call this method and what it allows to do here ? In addition, is there a follow-up action to make this method publicly accessible in the SDK if there are other similar use-cases ?

Reply:
Great catch. Comment added in he code of the `OpenTelemetrySdkService`. The use cases are:
* Debugging & troubleshooting helping find traces if they are missing reporting the `service.name` and `service.namespace`
* Testing: before this PR, we misconfigured the AutoConfigureOtlSdk and overwriting resource attributes through system properties or env vars failed.

> In addition, is there a follow-up action to make this method publicly accessible in the SDK if there are other similar use-cases ?
Valid point, I didn't bring thes us cases to the Java SIG as I did in the past for other SDK config topics. We have the same debugging/troubleshooting use case in the Jenkins OTel plugin.